.ci/scripts/test_synapse_port_db.sh

```diff
@@ -45,9 +45,23 @@ rm .ci/test_db.db
 poetry run update_synapse_database --database-config .ci/sqlite-config.yaml --run-background-updates
 
 # re-create the PostgreSQL database.
-poetry run .ci/scripts/postgres_exec.py \
-  "DROP DATABASE synapse" \
-  "CREATE DATABASE synapse"
+psql \
+  -c "DROP DATABASE synapse" \
+  -c "CREATE DATABASE synapse"
 
 echo "+++ Run synapse_port_db against empty database"
 poetry run synapse_port_db --sqlite-database .ci/test_db.db --postgres-config .ci/postgres-config.yaml
+
+echo "--- Create a brand new postgres database from schema"
+cp .ci/postgres-config.yaml .ci/postgres-config-unported.yaml
+sed -i -e 's/database: synapse/database: synapse_unported/' .ci/postgres-config-unported.yaml
+psql -c "CREATE DATABASE synapse_unported"
+poetry run update_synapse_database --database-config .ci/postgres-config-unported.yaml --run-background-updates
+
+echo "+++ Comparing ported schema with unported schema"
+# Ignore the tables that portdb creates. (Should it tidy them up when the porting is completed?)
+psql synapse -c "DROP TABLE port_from_sqlite3;"
+pg_dump --format=plain --schema-only --no-tablespaces --no-acl --no-owner synapse_unported > unported.sql
+pg_dump --format=plain --schema-only --no-tablespaces --no-acl --no-owner synapse          >   ported.sql
+# By default, `diff` returns zero if there are no changes and nonzero otherwise
+diff -u unported.sql ported.sql | tee schema_diff
```

Comment:
Obviously this is a syntactic diff. Do we have reasonable confidence that this will be OK for semantically equivalent databases?

(does `pg_dump` have a deterministic dump order etc?)

Reply:
The schema is apparently dumped in a deterministic order, see https://www.postgresql.org/message-id/1303223266.24799.13.camel%40fsopti579.F-Secure.com. https://stackoverflow.com/a/2179376/5252017 claims it's dumped in a "fairly deterministic" order.

No guarantees about the data. I suggest we leave this be for now: there's very few rows inserted into a brand new synapse database---IIRC they're just stream positions and schema metadata.

Reply:
Oh, I only dump the schema in the PR anyway.

I suppose we could also dump data and check they agree, but  I'm tempted to leave this PR as is.  WDYT?